ovs/lib/albacontroller.py

```diff
@@ -1447,6 +1453,174 @@ def unlink_alba_backends(target_guid, linked_guid):
         parent.invalidate_dynamics()
         parent.backend.invalidate_dynamics()
 
+    @staticmethod
+    @celery.task(name='alba.checkup_maintenance_agents', schedule=crontab(minute='0', hour='*'))
+    @ensure_single(task_name='alba.checkup_maintenance_agents', mode='CHAINED')
+    def checkup_maintenance_agents():
+        """
+        Check if requested nr of maintenance agents / backend is actually present
+        Add / remove as necessary
+        :return: None
+        """
+        service_template_key = 'alba-maintenance_{0}-{1}'
+
+        def _generate_name(_backend_name):
+            unique_hash = ''.join(random.choice(string.ascii_letters + string.digits) for _ in range(16))
+            return service_template_key.format(_backend_name, unique_hash)
+
+        def _count(_service_map):
+            amount = 0
+            for _services in _service_map.values():
+                amount += len(_services)
+            return amount
+
+        def _add_service(_node, _service_name, _abackend):
+            _name = _abackend.backend.name
+            try:
+                _node.client.add_maintenance_service(name=_service_name,
+                                                     alba_backend_guid=_abackend.guid,
+                                                     abm_name=AlbaController.get_abm_service_name(_abackend.backend))
+                return True
+            except Exception:
+                AlbaController._logger.exception('Could not add maintenance service for {0} on {1}'.format(_name, _node.ip))
+            return False
+
+        def _remove_service(_node, _service_name, _abackend):
+            _name = _abackend.backend.name
+            try:
+                _node.client.remove_maintenance_service(name=_service_name)
+                return True
+            except Exception:
+                AlbaController._logger.exception('Could not remove maintenance service for {0} on {1}'.format(_name, _node.ip))
+            return False
+
+        AlbaController._logger.info('Loading maintenance information')
+        service_map = {}
+        node_load = {}
+        available_node_map = {}
+        all_nodes = []
+        for node in AlbaNodeList.get_albanodes():
+            try:
```

Comment:
Too broad try-except clause imo
All of the code in it should work all the time, except for the ```node.client``` calls
Only those should be in a try-except
If an error occurs in another part of the code, the code should raise because other issues are present then and in current implementation, we would have 0 maintenance agents and no useful information would be in the logging